### ISSUE TITLE
CodePush doesn't sync on App start

### ISSUE THREAD

**Issue by danielbm** [Microsoft/react-native-code-push#939]
### Description

By using the default settings, the app doesn't check for updates. It has to be made explicitly by using codePush.sync().

### Reproduction

I've just used the default settings on installation, with the basic plugin usage:

```
import codePush from "react-native-code-push";

export default class App extends React.Component {
(...)
}

App = codePush(App);
```

### Additional Information

* react-native-code-push version: 3.0.1-beta | 2.1.1-beta (same issue with both)
* react-native version: 0.45.1
* Android version: Android 6.0.1

I debugged the code, and put breakpoints on _CodePush.js_. It turns out that it never reaches line 485:

`CodePush.sync(options, syncStatusCallback, downloadProgressCallback, handleBinaryVersionMismatchCallback);`

which I assume makes the synchronization.

Things work fine when I call `codePush.sync()` on the App.

**Comment by max-mironov** [comment]
Hi @danielbm thanks for your question. 

So I have tried your case but I could successfully update my app. Here are my repro steps:
1) Create new rn app with `rninit init RNISSUE939 --source react-native@0.45.1`
2) `cd MyApp`
3) Add code-push plugin with `npm install --save react-native-code-push@3.0.1-beta`
4) Link plugin with `react-native link react-native-code-push`
5) Add my android Deployment key while linking
6) Code-pushify my index.android.js as follows:
```
import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View
} from 'react-native';

import codePush from "react-native-code-push";

export default class RNISSUE939 extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.android.js
        </Text>
        <Text style={styles.instructions}>
          Double tap R on your keyboard to reload,{'\n'}
          Shake or press menu button for dev menu
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

RNISSUE939 = codePush(RNISSUE939);

AppRegistry.registerComponent('RNISSUE939', () => RNISSUE939);
```
7) Deploy the app to the device 
8) Verify it was running successfully
9) Release code-push update, restart the app to force it to check/dl update, restart it again to apply update
10) Verify update was applied
11) Double check that updates were reported successfully to service with `code-push deployment history test-r-android Staging` command
12) Verified the output:
![image](https://user-images.githubusercontent.com/24542698/28617650-0b4e8984-71b6-11e7-8a6a-0ccd39de2be5.png)


Please let us know if you have another steps to reproduce it or if we are missing something.

**Comment by max-mironov** [comment]
Hi again @danielbm, just a friendly ping on this ;)

**Comment by danielbm** [comment]
Hi @max-mironov ! I followed your steps, and everything worked fine. I will try to think what I did differently on my code on the first time. Maybe something related to the order that I installed code-push versions (I installed version 4.0.0, called react native link, then installed 2.1.1, and finally 3.0.0).

I will close the issue for now, and re-open if I am able to recreate the steps to reproduce.
